x-pack/plugins/actions/server/builtin_action_types/case/utils.ts

```diff
@@ -0,0 +1,253 @@
+  return normalizeMapping(SUPPORTED_SOURCE_FIELDS, mapping).reduce((fieldsMap, field) => {
+    const { source, target, actionType } = field;
+    fieldsMap.set(source, { target, actionType });
+    fieldsMap.set(target, { target: source, actionType });
+    return fieldsMap;
+  }, new Map());
+};
+
+export const mapParams = (
+  params: Partial<ExecutorActionParams>,
+  mapping: Map<string, MapRecord>
+): AnyParams => {
+  return Object.keys(params).reduce((prev: AnyParams, curr: string): AnyParams => {
+    const field = mapping.get(curr);
+    if (field) {
+      prev[field.target] = params[curr];
+    }
+    return prev;
+  }, {} as AnyParams);
+};
+
+export const createConnectorExecutor = ({
+  api,
+  createExternalService,
+}: CreateConnectorBasicArgs) => async (
+  execOptions: ActionTypeExecutorOptions
+): Promise<ActionTypeExecutorResult> => {
+  const actionId = execOptions.actionId;
+  const {
+    casesConfiguration: { mapping: configurationMapping },
+  } = execOptions.config as ConnectorPublicConfigurationType;
+
+  const params = execOptions.params as ExecutorParams;
+  const { subAction, subActionParams } = params;
+  const { comments, externalId, ...restParams } = subActionParams;
+
+  const mapping = buildMap(configurationMapping);
+  const externalCase = mapParams(restParams as ExecutorActionParams, mapping);
+
+  const res: Pick<ActionTypeExecutorResult, 'status'> &
+    Pick<ActionTypeExecutorResult, 'actionId'> = {
+    status: 'ok',
+    actionId,
+  };
+
+  const externalService = createExternalService({
+    config: execOptions.config,
+    secrets: execOptions.secrets,
+  });
+
+  if (!api[subAction]) {
+    throw new Error('[Action][Connector] Unsupported subAction type');
+  }
+
+  const data = await api[subAction]({
+    externalService,
+    mapping,
+    params: { ...subActionParams, externalCase },
+  });
+
+  return {
+    ...res,
+    data,
+  };
+};
+
+export const createConnector = ({
+  api,
+  config,
+  validate,
+  createExternalService,
+  validationSchema,
+}: CreateConnectorArgs) => {
+  return ({
+    configurationUtilities,
+    executor = createConnectorExecutor({ api, createExternalService }),
+  }: CreateActionTypeArgs): ActionType => ({
+    id: config.id,
+    name: config.name,
+    minimumLicenseRequired: 'platinum',
+    validate: {
+      config: schema.object(validationSchema.config, {
+        validate: curry(validate.config)(configurationUtilities),
+      }),
+      secrets: schema.object(validationSchema.secrets, {
+        validate: curry(validate.secrets)(configurationUtilities),
+      }),
+      params: ExecutorParamsSchema,
+    },
+    executor,
+  });
+};
+
+export const throwIfNotAlive = (
+  status: number,
+  contentType: string,
+  validStatusCodes: number[] = [200, 201, 204]
+) => {
+  if (!validStatusCodes.includes(status) || !contentType.includes('application/json')) {
+    throw new Error('Instance is not alive.');
+  }
+};
+
+export const request = async ({
```

Comment:
I think we should use a generic type here so we can define it if we need and will simplify our types and be cleaner
```
export const request = async <T = unknown>({
  axios,
  url,
  method = 'get',
  data,
}: {
  axios: AxiosInstance;
  url: string;
  method?: Method;
  data?: T;
}): Promise<AxiosResponse> => {
  const res = await axios(url, { method, data: data ?? {} });
  throwIfNotAlive(res.status, res.headers['content-type']);
  return res;
};
```

Reply:
I will do the same for patch